corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

```diff
@@ -682,6 +682,56 @@ def requirejs_main(parser, token):
     ]))
 
 
+@register.tag
+def webpack_main(parser, token):
+    """
+    Indicate that a page should be using Webpack, by naming the
+    JavaScript module to be used as the page's main entry point.
+
+    The base template need not specify a value in its `{% webpack_main %}`
+    tag, allowing it to be extended by templates that may or may not
+    use requirejs. In this case the `webpack_main` template variable
+    will have a value of `None` unless an extending template has a
+    `{% webpack_main "..." %}` with a value.
+    """
+    bits = token.contents.split(None, 1)
+    if len(bits) == 1:
+        tag_name = bits[0]
+        value = None
+    else:
+        tag_name, value = bits
+
+    if getattr(parser, "__saw_webpack_main", False):
+        raise TemplateSyntaxError(
+            "multiple '%s' tags not allowed (%s)" % tuple(bits))
+    parser.__saw_webpack_main = True
+
+    if value and (len(value) < 2 or value[0] not in '"\'' or value[0] != value[-1]):
+        raise TemplateSyntaxError("bad '%s' argument: %s" % tuple(bits))
+
+    # use a block to allow extension template to set requirejs_main for base
+    return loader_tags.BlockNode("__" + tag_name, NodeList([
+        WebpackMainNode(tag_name, value and value[1:-1])
+    ]))
+
+
+class WebpackMainNode(template.Node):
+
+    def __init__(self, name, value):
+        self.name = name
+        self.value = value
+
+    def __repr__(self):
+        return "<RequireJSMain Node: %r>" % (self.value,)
```

Comment:
Suggesting here because I can't below on `RequireJSMainNode`, but it should actually be applied down there.

```suggestion
        return f"<{type(self).__name__} Node: {self.value!r}>"
```